polkadot-parachains/parachains-common/src/impls.rs

```diff
@@ -98,14 +98,41 @@ where
 	}
 }
 
+/// Tests `loc` whether it contains `prefix`.
+/// Similar to `MultiLocation::match_and_split` but allows arbitrary suffixes (instead of just one
+/// junction).
+///
+/// ```
+/// # use xcm::prelude::*;
+/// # use parachains_common::impls::match_prefix;
+/// let prefix =  MultiLocation::new(1, X1(Parachain(1234)));
+/// let loc =  MultiLocation::new(1, X2(Parachain(1234), GeneralIndex(5)));
+/// assert!(match_prefix(&prefix, &loc));
+/// let failing =  MultiLocation::new(1, Here);
+/// assert!(!match_prefix(&prefix, &failing));
+/// ```
+pub fn match_prefix(prefix: &MultiLocation, loc: &MultiLocation) -> bool {
```

Comment:
This looks like it's a function that should be upstreamed to Polkadot.

Reply:
yep, agree, as mentioned in the description :-D

Reply:
https://github.com/paritytech/polkadot/pull/4827